File: src/main/java/org/codeberg/zenxarch/zombies/data/ZBiomeTags.java
package org.codeberg.zenxarch.zombies.data;

import net.minecraft.registry.RegistryKeys;
import net.minecraft.registry.tag.TagKey;
import net.minecraft.world.biome.Biome;
import org.codeberg.zenxarch.zombies.Zombies;

public interface ZBiomeTags {
  public static final TagKey<Biome> WITHOUT_ZOMBIE_APOCALYPSE = of("without_zombie_apocalypse");
  public static final TagKey<Biome> WITH_FLAME_ZOMBIES = of("with_flame_zombies");
  public static final TagKey<Biome> WITH_FROST_ZOMBIES = of("with_frost_zombies");
  public static final TagKey<Biome> WITH_AXE_ZOMBIES = of("with_axe_zombies");
  public static final TagKey<Biome> WITH_SWAMP_ZOMBIES = of("with_swamp_zombies");

  private static TagKey<Biome> of(String id) {
    return TagKey.of(RegistryKeys.BIOME, Zombies.id(id));
  }
}

File: src/main/java/org/codeberg/zenxarch/zombies/data/ZBiomeTags.kt
package org.codeberg.zenxarch.zombies.data

import net.minecraft.registry.RegistryKeys
import net.minecraft.registry.tag.TagKey
import net.minecraft.world.biome.Biome
import org.codeberg.zenxarch.zombies.Zombies.id

object ZBiomeTags {
    val WITHOUT_ZOMBIE_APOCALYPSE: TagKey<Biome> = of("without_zombie_apocalypse")

    private fun of(id: String) = TagKey.of(RegistryKeys.BIOME, id(id))
}
